### ISSUE TITLE
Serious BUGs for MySQL- Look inside for details

### ISSUE THREAD

**Issue by ondras** [ondras/wwwsqldesigner#153]
_From [scrap...@gmail.com](https://code.google.com/u/114909813515663524560/) on January 05, 2012 19:14:15_

OK, I was thinking this was a very nice tool, but have since found some very SERIOUS bugs that make it unreliable for complete usage.

Look below and save yourself LOTS of time and frustration!

1. MySQL 5.1.60 on shared webhost

2. wwwsqldesigner 2.6 on Windows 7, running from webhost (same as in `#1`) via Firefox 8.0

The MySQL code generated has a few problems.

1. Text fields use field size in parenthesis for create table, which will NOT work. Ex: collector (100)   \<--You MUST remove the (100) for it to work. In MySQL 5.1 (at least) the 100 is where collation and such would be. Syntax error.

2. I enabled the option to display field sizes in the table and then reloaded the page. When you output the code, the TEXT fields are created with the size as part of the name!!! Ex: Name: collector (100)
This causes a form not to submit to add_record.php (form input field must match field of database exactly). This was very difficult to find and only caught my attention after much debugging and using an admin program to view the table structure that was created.

So both issues seem related to Text type. I don't know if this is due to changes in mySQL with v5.1 or what, but as is WILL NOT WORK! You can modify the output code and use it, but it would be better to fix it in the program itself- PLEASE! :)

In summary, I like the simple and clean interface. It makes visualizing data very simple, but it really needs to be fixed so it functions properly with text fields in current versions of mySQL.

I like it. Please fix it.

- JC

_Original issue: http://code.google.com/p/wwwsqldesigner/issues/detail?id=146_

**Comment by ondras** [comment]
_From [ondrej.zara](https://code.google.com/u/ondrej.zara/) on January 05, 2012 11:39:44_

The issue `#2` (field size as a part of field name) was already fixed ( https://code.google.com/p/wwwsqldesigner/source/detail?r=b06737612b95517be74ec68467b69bbb66aa9ca6 ), please update your wwwsqldesigner.js to latest revision.

I will further investigate issue `#1` and let you know the resolution.

**Owner:** ondrej.zara

**Comment by ondras** [comment]
_From [ondrej.zara](https://code.google.com/u/ondrej.zara/) on February 10, 2012 05:12:31_

The generated code should work:

mysql> create table test ( a char (10), b varchar (20) );
Query OK, 0 rows affected (0.06 sec)

$ dpkg -l | grep "mysql-server\s"
ii  mysql-server                                            5.1.49-3

**Comment by ondras** [comment]
_From [ondrej.zara](https://code.google.com/u/ondrej.zara/) on March 15, 2014 02:18:20_

Closing for no activity.

**Status:** Done
